integrations/lib/multiwoven/integrations/destination/zendesk/client.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+module Multiwoven
+  module Integrations
+    module Destination
+      module Zendesk
+        include Multiwoven::Integrations::Core
+        class Client < DestinationConnector
+          prepend Multiwoven::Integrations::Core::RateLimiter
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            initialize_client(connection_config)
+            authenticate_client
+            success_status
+          rescue StandardError => e
+            handle_exception("ZENDESK:CHECK_CONNECTION:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def discover(_connection_config = nil)
+            catalog = build_catalog(load_catalog)
+            catalog.to_multiwoven_message
+          rescue StandardError => e
+            handle_exception("ZENDESK:DISCOVER:EXCEPTION", "error", e)
+            failure_status(e)
+          end
+
+          def write(sync_config, records, action = "create")
+            @action = sync_config.stream.action || action
+            initialize_client(sync_config.destination.connection_specification)
+            process_records(records, sync_config.stream)
```

Comment:
write method should return tracking_message(write_success, write_failure)

Reply:
added.